proposals/accepted/047-standard-library-reform.rst

```diff
@@ -0,0 +1,559 @@
+`This PR <https://github.com/rust-lang/rust/pull/63676>`_ made that change, using the ``wasi`` library (Rust bindings to WASI system calls) directly.
+This is what we should emulate in order to provide a top-tier programming environment for greenfield WebAssembly applications in Haskell.
+
+Prior attempts at splitting ``base``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There have been attempts to split ``base`` before, but they attempted to get everything done at once, setting a dangerously high bar for success.
+This approach here, by contrast, first and foremost seeks to avoid those difficulties and find a sustainable, suitably low-risk approach.
+It is much more concerned with how we safely approach these issues than what the exact outcome looks like.
+
+Technical Content
+-----------------
+
+Here is a plan to solve these issues.
+
+**Step 1A**: Task the CLC with defining new standard libraries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Based on the conversation in `CLC Issue #015`_, ``base`` is exposing too much stuff, yet trying to limit what is exposed would be a big breaking change.
+
+The solution is to reach for another layer of indirection.
+The CLC should be tasked with devising new standard library interfaces, which would initially be implemented by reexporting modules from ``base``.
+
+The new library interfaces should be carefully designed in and of themselves to tackle many, but not all, of the issues above:
+
+- They should be designed *not* to break every release.
+  Even though the underlying ``base`` from which modules are reexported would continue to have its regular problematic major version bumps, the portion reexported should have very infrequent breaking changes.
+
+  This fixes **Problem 1**.
+
+- These libraries should be emphasized in all documentation, and users should be encouraged to use them and not ``base`` in new end-application code.
+  ``base``, in contrast, would be kept exposed as a mere legacy interface.
+  As code migrates over to use the new standard libraries, ``base`` should become less important.
+  GHC devs can therefore feel increasingly confident modifying parts of ``base`` which are *not* reexported in these new libraries.
+
+  This partially fixes **Problem 2**.
+
+- The new standard library should not be a single library but multiple.
+  IO-free interfaces that are portable everywhere should be one library.
+  Interfaces involving IO should be split into libraries where they run.
+
+  For example, Unix and Windows are mostly a superset of WASI, so WASI-compatible file-descriptor-oriented code should work everywhere.
+
+  Exactly how many separate libraries is justified is left to the CLC to decide.
+
+  This fixes **Problem 3**.
+
+- Because these are new libraries "on top" of ``base``, they can also reexport items from libraries, like ``text``.
+  The CLC should consider such reexports.
+
+  This fixes **Problem 5**.
+
+New Goal: Rationalize dependencies
```

Comment:
I don't know why this is a new goal, rather than a technique to solve Problems 2 and 4. In any case, I also don't know what the goal is here. Can you provide a description of what it would mean to rationalize the dependencies?